src/gromacs/nblib/integrator.cpp

```diff
@@ -74,9 +74,7 @@ void LeapFrog::integrate(const real dt)
             x[i][dim] += v[i][dim] * dt;
         }
     }
-    matrix box;
-    gmx::fillLegacyMatrix(simulationState_.box().matrix(), box);
-    put_atoms_in_box(PbcType::Xyz, box, x);
+    put_atoms_in_box(PbcType::Xyz, simulationState_.box().legacyMatrix, x);
```

Comment:
This is a user-facing function. The fact that there's a legacyMatrix inside box should be hidden from the user
by providing a wrapper.

Reply:
The fact that it is a C-array that has to be copied in place means that the solutions are inelegant. Either we expose a member, or return a pointer. 

It is a short-lived interim solution. The moment GROMACS moves to using the new `Matrix` class, this aspect of box will get a lot cleaner.

Reply:
I don't think this will be such a short-lived solution. Unless you have a plan to refactor all of the PBC code in gromacs to use Matrix, we will have to live with the legacy matrix for quite some time.

Reply:
Done. Turned into a wrapper.